site/src/pages/docs/client-retry.mdx

```diff
@@ -71,7 +71,8 @@ new RetryRule() {
             }
         }
 
-        if (ctx.log().responseHeaders().status() == HttpStatus.TOO_MANY_REQUESTS) {
+        HttpStatus status = ctx.log().ensureAvailable(RequestLogProperty.RESPONSE_HEADERS).responseHeaders().status();
+        if (status == HttpStatus.TOO_MANY_REQUESTS) {
```

Comment:
I don't think we need to do this because the response headers are always available at this point. It can be unavailable only when `cause != null`.

Reply:
Ah, looks like we can reach here even when `cause != null`. I think we should guard with `isAvailable()` instead of using `ensureAvailable()`, so that `shouldRetry()` does not raise an exception but just return `next()`.

Reply:
> It can be unavailable only when cause != null.

Well, it's available even when `cause != null`. (With the status `HttpStatus.UNKNOWN`)
So I think we can use `partial()` which was used at first. Sorry about it. 😅

Reply:
Update back!